### PR TITLE
Daemons initilization locks

### PR DESCRIPTION
**Pull request by heitorPB**
depends on https://github.com/omnivector-solutions/slurm-ops-manager/pull/61

### CODE REVIEW

File: charm-slurmctld/src/charm.py [hunk 1]
```diff
@@ -151,13 +151,19 @@ def _on_install(self, event):
             # NOTE: Use leadership settings instead of stored state when
             # leadership settings support becomes available in the framework.
             if self._is_leader():
+                # NOTE the backup controller should also have the jwt and munge
```
Comment (jamesbeedy) [inline charm-slurmctld/src/charm.py:154]: 👍

File: charm-slurmctld/src/charm.py [hunk 2]
```diff
@@ -293,23 +310,29 @@ def _on_write_slurm_config(self, event):
...
-            # update down nodes cache
             self._stored.down_nodes = down_nodes.copy()
 
+            # NOTE: does it make sense to do this here?
             # slurmrestd needs the slurm.conf file, so send it
             if self._stored.slurmrestd_available:
                 self._slurmrestd.set_slurm_config_on_app_relation_data(
                     slurm_config
                 )
+                # NOTE: check if we really need to restart slurmrestd. scontrol
```
Comment (jamesbeedy) [inline charm-slurmctld/src/charm.py:334]: 👍

Comment (jamesbeedy) [reply charm-slurmctld/src/charm.py:334]: let make sure and figure this out in the rest of the slurmrestd implementation

**Review by jamesbeedy — APPROVED** [review]
👍

**Review by matheushent — COMMENTED** [review]
👍🏻

**Pull request merged**